basictracer/span.py

```diff
@@ -49,6 +51,26 @@ def log_kv(self, key_values, timestamp=None):
             self.logs.append(LogData(key_values, timestamp))
         return super(BasicSpan, self).log_kv(key_values, timestamp)
 
+    def capture(self):
+        with self._lock:
+            self._refcount += 1
+
+    def deactivate(self):
+        # TODO: this safe-guard breaks the case of start_manual_span()
+        # with a context manager; probably we should refactor this approach
+        # because may not fit very well with the Python API. Another
+        # approach is using this check only to decide if calling
+        # `make_active(self._to_restore)` or not.
```

Comment:
Correct. I was thinking about the `start_manual_span` case. So, yes, an option is to keep state, internally (or publicly even) about whether the `Span` is marked as active or not (i.e. `is_active` property, False for `start_manual_span`, True for `start_active_span`), and based on it, go one way or another.

I outlined a few thoughts/approaches we may take here https://gist.github.com/carlosalberto/4c15315a94dafe8059d620e000680954.

Else, remember doing an `ActiveSpan` just like Java is still an option. For your consideration.